spi/data-plane-selector/data-plane-selector-spi/src/main/java/org/eclipse/edc/connector/dataplane/selector/spi/instance/DataPlaneInstance.java

```diff
@@ -14,53 +14,172 @@
  * and a timestamp of its last selection time. In addition, there are extensible properties to hold specific properties.
  */
-@JsonSubTypes({
-        @JsonSubTypes.Type(value = DataPlaneInstanceImpl.class, name = "dataspaceconnector:dataplaneinstance")
-})
-public interface DataPlaneInstance extends Polymorphic {
+public class DataPlaneInstance {
 
-    /**
-     * Gets the unique identifier for this instance
-     */
-    String getId();
+    private Map<String, Object> properties;
+
+    @JsonProperty("allowedSourceTypes")
+    private Set<String> allowedSourceTypes;
+
+    @JsonProperty("allowedDestTypes")
+    private Set<String> allowedDestTypes;
+
+    private int turnCount;
+
+    private long lastActive;
+
+    private URL url;
+
+    private String id;
+
+    protected DataPlaneInstance() {
+        turnCount = 0;
+        lastActive = Instant.now().toEpochMilli();
+        properties = new HashMap<>();
+        url = null;
+
+        allowedSourceTypes = new HashSet<>();
+        allowedDestTypes = new HashSet<>();
+    }
+
+    public String getId() {
+        return id;
+    }
 
     /**
-     * Determines whether a particular DPF instance can handle the given request.
+     * Determines whether this instance can handle a particular source and data address, by evaluating {@link DataAddress#getType()}
+     * against an internal list of allowed source and dest types.
      *
      * @param sourceAddress      The location where the data is located
      * @param destinationAddress The destination address of the data
+     * @return true if can handle, false otherwise.
      */
-    boolean canHandle(DataAddress sourceAddress, DataAddress destinationAddress);
+    public boolean canHandle(DataAddress sourceAddress, DataAddress destinationAddress) {
+        Objects.requireNonNull(sourceAddress, "source cannot be null!");
+        Objects.requireNonNull(destinationAddress, "destination cannot be null");
+        return allowedSourceTypes.contains(sourceAddress.getType()) && allowedDestTypes.contains(destinationAddress.getType());
+    }
 
-    /**
-     * returns the url (host+path) to the DataPlane's control API
-     * Note that this API is not intended to be public and may require authentication.
-     */
-    URL getUrl();
+    public URL getUrl() {
+        return url;
+    }
 
-    /**
-     * Indicates how often one particular DPF instance has been selected
-     */
-    int getTurnCount();
+    public int getTurnCount() {
+        return turnCount;
+    }
 
-    /**
-     * The last time a particular instance was selected in POSIX time. If it has never been selected before, this returns 0.
-     */
-    long getLastActive();
+    public long getLastActive() {
+        return lastActive;
+    }
 
-    /**
-     * A list of extensible properties, for example this could contain a DPF's public-facing API.
-     */
-    Map<String, Object> getProperties();
+    public Map<String, Object> getProperties() {
+        return properties;
```

Comment:
maybe this should be immutable: `return Collections.unmodifiableMap(properties)`?